tests/system/src/test/java/test/robot/javafx/scene/NodeInitializationStressTest.java

```diff
@@ -408,9 +407,14 @@ public void pagination() {
             c.setPageFactory((pageIndex) -> {
                 return new Label(pageIndex + " " + nextString());
             });
-            c.setPageCount(100);
-            c.setCurrentPageIndex(nextInt(100));
+            int mx = 1 + nextInt(100);
             accessControl(c);
+            c.setPageCount(mx);
+            c.setCurrentPageIndex(nextInt(mx));
+            // this test generates a lot of garbage quickly
+            if (nextBoolean(0.1)) {
+                System.gc();
+            }
```

Comment:
While investigation the Pagination leak I also took a closer look at this `System.gc()`. The reason this prevents the OOM has nothing to do with the GC per se, but rather it slows the loops down so much that they don't generate nearly as much garbage in the 5 seconds the test runs in. You can verify that yourself by replacing the `System.gc()` with `sleep(10)` or similar (depending on the speed of your system), and get the same effect.

I don't think there is a good reason to call `System.gc()` at all in this loop. Once the leak is fixed, it  should run fine at full speed.